pkg/controller/admissionchecks/provisioning/controller.go

```diff
@@ -438,17 +443,27 @@ func requestHasParameters(req *autoscaling.ProvisioningRequest, prc *kueue.Provi
 	if req.Spec.ProvisioningClassName != prc.Spec.ProvisioningClassName {
 		return false
 	}
-	if len(req.Spec.Parameters) != len(prc.Spec.Parameters) {
-		return false
-	}
-	for k, vReq := range req.Spec.Parameters {
-		if vCfg, found := prc.Spec.Parameters[k]; !found || vReq != autoscaling.Parameter(vCfg) {
+	for k, vCfg := range prc.Spec.Parameters {
+		if vReq, found := req.Spec.Parameters[k]; !found || vReq != autoscaling.Parameter(vCfg) {
 			return false
 		}
 	}
 	return true
 }
 
+// passProvReqParams extracts from Workload's annotations ones that should be passed to ProvisioningRequest
+func passProvReqParams(wl *kueue.Workload, req *autoscaling.ProvisioningRequest) {
+	if req.Spec.Parameters == nil {
+		req.Spec.Parameters = make(map[string]autoscaling.Parameter, 0)
+	}
+	for annotation, val := range wl.Annotations {
+		if strings.HasPrefix(annotation, provReqAnnotationPrefix) {
+			paramName := strings.Split(annotation, "/")[1]
```

Comment:
you may consider https://pkg.go.dev/strings#TrimPrefix

Reply:
Thank you, I'll fix that